gossipd/gossipd.c

```diff
@@ -1809,6 +1811,53 @@ static struct io_plan *handle_payment_failure(struct io_conn *conn,
 	return daemon_conn_read_next(conn, daemon->master);
 }
 
+/*~ lightningd tells us when about a gossip message directly, when told to by
+ * the addgossip RPC call.  That's usually used when a plugin gets an update
+ * returned in an payment error. */
+static struct io_plan *inject_gossip(struct io_conn *conn,
+				     struct daemon *daemon,
+				     const u8 *msg)
+{
+	u8 *goss;
+	const u8 *errmsg;
+	const char *err;
+
+	if (!fromwire_gossipd_addgossip(msg, msg, &goss))
+		master_badmsg(WIRE_GOSSIPD_ADDGOSSIP, msg);
+
+	switch (fromwire_peektype(goss)) {
+	case WIRE_CHANNEL_ANNOUNCEMENT:
+		errmsg = handle_channel_announcement_msg(daemon, NULL, goss);
```

Comment:
Had to verify that this still performs the `chain_hash` check, which is good, otherwise we could end up with some botched experiments :-)